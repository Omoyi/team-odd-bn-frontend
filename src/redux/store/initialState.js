
const state = {
  search: {
    key: '',
    payload: '',
  },
  popUpsDisplay: {
    deleteComment: 'none',
    editTrip: 'none',
    currentPopUp: 'editTrip',
  },
  pagination: {
    pageNo: 0,
    itemsPerPage: 5,
  },
  accommodation: {
    createAccommodation: {
      rooms: [],
      photos: [],
    },
  },
  auth: {
    signup: {
      spinnerStatus: false,
      signupInputData: {},
      signupData: {},
      signupError: {},
    },
    signin: {
      spinnerStatus: false,
      signinInputData: {},
      signinData: {},
      signinError: {},
    },
  },
  allUsers: {
    paginatedRequest: [],
    paginationStart: 0,
    paginationEnd: 5,
  },
  trips: {
    tripRequests: {
      getCity: [],
      onewayData: {},
      onewayError: {},
      onewayInput: {},
    },
    availRequests: {
      approveRequestMessage: {},
    },
    requests: {
      requestsData: [],
      singleRequestData: {},
      requestCommentsData: {},
      requestsError: {},
      paginatedRequest: [],
      paginationStart: 0,
      paginationEnd: 5,
    },
    getStats: {
      getStatsInput: {
        tripType: 1,
      },
      getStatsData: {},
      getStatsError: {},
      tripStatsCounter: {},
    },
  },
  viewProfile: {
    profile: {},
  },
  requestsData: {
    data: '',
  },
};
export default state;
