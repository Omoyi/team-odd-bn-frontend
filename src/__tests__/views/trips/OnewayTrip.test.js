import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount, shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import stores from '../../../redux/store';
import OnewayTrip from '../../../containers/TripsContainer/oneway';
import initialState from '../../../redux/store/initialState';
import {
  redirectInitialState1,
  spinnerInitialState1,
} from '../../../__mocks__/auth/authMock';

const mockStore = configureStore([]);
let wrapper;
let store;

describe('On Oneway page', () => {
  it('Test spinner action', () => {
    store = mockStore(spinnerInitialState1);

    wrapper = shallow(
      <Router>
        <OnewayTrip
          store={store}
          handleSubmit={async () => {
            jest.fn();
            return jest.fn();
          }}
        />
      </Router>,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Spinner').length).toEqual(0);
  });
  it('Test snapshot using the initial state', () => {
    store = mockStore(initialState);

    wrapper = shallow(
      <Router>
        <OnewayTrip
          store={store}
          handleSubmit={async () => {
            jest.fn();
            return jest.fn();
          }}
        />
      </Router>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    wrapper = mount(
      <Provider store={stores}>
        <Router>
          <OnewayTrip
            handleSubmit={async () => {
              jest.fn();
              return jest.fn();
            }}
          />
        </Router>
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
