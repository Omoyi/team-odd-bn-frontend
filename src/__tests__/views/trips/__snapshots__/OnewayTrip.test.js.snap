// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On Oneway page Test snapshot using the initial state 1`] = `ShallowWrapper {}`;

exports[`On Oneway page Test spinner action 1`] = `ShallowWrapper {}`;

exports[`On Oneway page renders without crashing 1`] = `ReactWrapper {}`;
